labs/b4.md

```diff
@@ -0,0 +1,269 @@
+In this class, we will be focused on using Debian. As noted before, Debian uses
+`apt` or `dpkg` as its package manager.
+
+We invoke the package manager by using `apt`. Before installing anything, you
+shoud update your local copy of packages and versions to ensure the package
+manager installs the latest and most up-to-date packages. To do that, run:
+
+    $ sudo apt update
+
+To find the package to install:
+
+    $ apt search [package|description]
+
+To install a package, run:
+
+    $ sudo apt install package
+
+To remove a package, run:
+
+    $ sudo apt remove package
+
+Easy? When you want to upgrade the packages that you have installed when
+new versions are released, you can do so by calling:
+
+    $ sudo apt upgrade
+
+There are also other commands, such as removing unneeded dependencies (`sudo
+apt autoremove`) and purging packages (`sudo apt purge`) but that is what the
+man pages are for.
+
+### Example
+
+We are going to install GCC for the next step of the lab.
+
+**This, and all other parts of the lab, should be done on your DigitalOcean
+VMs, as you cannot install packages on OCF machines.**
+
+Please note that you can only connect to your VM from inside the Berkeley
+network, so you will have to either be on campus wifi, or you will have to SSH
+through `ssh.ocf.berkeley.edu` (or a similar on-campus host) to access your VM.
+You should be able to connect to it at `<OCF username>.decal.xcf.sh` by running
+`ssh <username>@<username>.decal.xcf.sh` and the password you set when your
+first logged in, or using the email you received.
+
+Simply run:
+
+    $ sudo apt install gcc
+
+Now check if GCC is installed by running the following to check GCC's version:
+
+    $ gcc --version
+
+Now install the `ruby-dev` and `ruby-ffi` packages for the next parts of the
+lab.
+
+## But what about software that isn’t in the repositories?
+
+Sometimes it happens that a program you want to install hasn't been
+packaged for your distributions, or hasn't been packaged at all. You have
+several options in this scenario to install the software you want:
+
+#### Warning:
+**Installing software not from a repository carries the same risks as
+installing software from a random .exe or .msi on Windows.**
+
+Linux is popularly considered more secure than Windows or MacOS because of
+its use of repositories -- but a malicious package installed manually can
+pwn your system as easily as malware on any other OSes.
+**Use common sense! Only manually install from sources you trust.**
+
+*(For users of Ubuntu: PPAs, or third-party repositories, carry the same
+risks of running malicious third-party code.)*
+
+If the developers provide a package compatible with your distribution (.deb for
+Debian-based distros), you can download that package and install it using:
+
+    $ sudo apt install ./package.deb
+
+Another way is to find a generic binary package from the developer. This can
+come in the form of shady shell scripts, or a binary tarball that you can just
```

Comment:
Explain what a tarball is maybe? Footnote is probably the best option since tarballs have briefly been mentioned before